server/src/session/index.ts

```diff
@@ -0,0 +1,70 @@
+import { createConnection, IConnection, TextDocuments } from 'vscode-languageserver';
+import {
+    DidChangeConfigurationParams,
+    Disposable,
+    IPCMessageReader,
+    IPCMessageWriter,
+    TextDocumentChangeEvent,
+} from 'vscode-languageserver-protocol';
+
+import { YmlCompletionItemsProvider } from '../completion/YmlCompletionItemsProvider';
+import { YmlDefinitionProvider } from '../definitions';
+import * as validateTextDocument from '../feature';
+import { ISettings } from '../interfaces';
+
+export default class Session implements Disposable {
```

Comment:
I admit I don’t quite get this concept of “session”. For example, what is a session’s scope? Is there one session per window? Etc.
Consider adding a few words of documentation for dummies like me.
It seems also slightly strange to me to have direct access through that class to the `warning`, etc. logging methods. I don’t see sessions printing stuff that often. :thinking: Oh, that seems to be what this OCaml project does. But what made you hold that project in such high esteems, to begin with? :thinking: